graph/community/louvain_directed.go

```diff
@@ -353,25 +352,25 @@ func (g *ReducedDirected) To(v graph.Node) []graph.Node {
 func (g *ReducedDirected) HasEdgeBetween(x, y graph.Node) bool {
 	xid := x.ID()
 	yid := y.ID()
-	if xid == yid {
+	if xid == yid || xid != int64(int(xid)) || yid != int64(int(yid)) {
```

Comment:
There are three things here. One is that the changes associated with moving from `int` to `int64` are not at all clearly cleaner or more readable (this was my first motivation for limiting the change - I did actually go through and do the complete transformation first and decided it was not the way to go) and the second is that there is an immediate ~10% improvement in performance available as I can partially revert the removal of intsets, which clearly did show an improvement in this context. The last is safety, community graphs are held in slices, so the appropriate ID type is an `int` internally.

I will, however, add an `isValidCommunityID` to replace the `x = int64(int(x))` checks - adding sign checks which I don't yet have.